libethashseal/EthashProofOfWork.h

```diff
@@ -47,14 +48,38 @@ struct EthashProofOfWork
 
 	struct WorkPackage
 	{
-		WorkPackage() = default;
+		WorkPackage() {}
 		WorkPackage(BlockHeader const& _bh);
-		void reset() { headerHash = h256(); }
-		operator bool() const { return headerHash != h256(); }
+		WorkPackage(WorkPackage const& _other)
```

Comment:
I'm not sure this will work, but try to get rid of copying it in `EthashCPUMiner::workLoop()` (the line `WorkPackage w = work();` - relpace it with const reference), then probably we can keep `WorkPackage` noncopyable